### PR TITLE
SCHEMA: Add sorted() and index() functions, update expression tests

### PR DESCRIPTION
**Pull request by effigies**
To enable some check implementations, we need sorted() and index(). Definitions and tests are provided. Checks will come in a separate PR.

### CODE REVIEW

**Review by bendhouseart — APPROVED** [review]
Sure.

File: src/schema/meta/expression_tests.yaml [hunk 1]
```diff
@@ -77,15 +84,61 @@
...
+- expression: count([1, 2, 3], 1)
+  result: 1
+- expression: index(["i", "j", "k"], "i")
+  result: 0
+- expression: index(["i", "j", "k"], "j")
+  result: 1
+- expression: index(["i", "j", "k"], "x")
+  result: null
+- expression: sorted([3, 2, 1])
+  result: [1, 2, 3]
```
Comment (tsalo) [inline src/schema/meta/expression_tests.yaml:136]: Are there any non-numeric cases for sorted in the spec?

Comment (effigies) [reply src/schema/meta/expression_tests.yaml:136]: The only use case we've found is for checking that certain arrays are monotonically increasing.

Comment (tsalo) [reply src/schema/meta/expression_tests.yaml:136]: awesome!

**Review by tsalo — APPROVED** [review]

**Pull request merged**